### DISCUSSION TITLE
Typing composable return

### DISCUSSION THREAD

**Discussion by danillo-hilbs** [nuxt/nuxt#17297] [Questions]
I'm new to typescript, and I would like to know how I can type the return of a composable...

![image](https://user-images.githubusercontent.com/34604307/178342970-b59ab621-9974-4f76-b76a-512fa3217f3c.png)

![image](https://user-images.githubusercontent.com/34604307/178343001-b1dda3c9-508b-4f7c-8498-8fdff6b80d46.png)

**Comment by mihail727** [comment]
some ways:::
1) change return of function
```js
return {
    dimensions.value,
    error.value,
    pending.value,
    refresh
}
```

2) change the return type of a function:
```js
:Promise<{
    dimensions: Ref<Dimension[],
    error: Ref<boolean | Error>,
    pending: Ref<boolean>,
    refresh: (opts?: RefreshOptions | undefined)>
}>
```

3) return directly:
```js
return useFetchApi(`${basePath}/${id}`)
```
